### PR TITLE
feat: add ability to register multiple sidechains on the base layer

### PR DESCRIPTION
**Pull request by stringhandler**
BREAKING CHANGE: Adds Shard to QC's and proposals. Removes merkle proof of VNS

### CODE REVIEW

File: dan_layer/consensus/src/block_validations.rs [hunk 1]
```diff
@@ -127,32 +127,40 @@ pub async fn check_quorum_certificate<TConsensusSpec: ConsensusSpec>(
     vote_signing_service: &TConsensusSpec::SignatureService,
     epoch_manager: &TConsensusSpec::EpochManager,
 ) -> Result<(), HotStuffError> {
-    if candidate_block.justify().epoch().as_u64() == 0 {
-        // Ignore genesis block.
+    let qc = candidate_block.justify();
```
Comment (stringhandler) [inline dan_layer/consensus/src/block_validations.rs:130]: @sdbondi Please have a look at this logic, I expanded it a bit and it's potentially out of scope of this PR

File: integration_tests/tests/features/claim_fees.feature [hunk 2]
```diff
@@ -84,52 +84,3 @@ Feature: Claim Fees
 
     # Claim fees into ACC2
     When I claim fees for validator VN and epoch 1 into account ACC2 using the wallet daemon WALLET_D, it fails
-
-  @serial
-  Scenario: Prevent validator fees claim for unauthorized wallet
```
Comment (stringhandler) [inline integration_tests/tests/features/claim_fees.feature:89]: I deleted this test as it was basically the same as the one above

Comment (sdbondi) [reply integration_tests/tests/features/claim_fees.feature:89]: This test above checks that the fees can't be double-claimed. This test checks if a wallet does not use/possess the private claim key, it cannot claim fees.

File: dan_layer/consensus/src/block_validations.rs (hunk 1 above)
Comment (sdbondi) [reply dan_layer/consensus/src/block_validations.rs:130]: Yeah the previous check isn't right. Also agree with the comment that a blanket "LGTM" for all genesis blocks in a number of places in the code is wrong. I think we need to refactor/audit the validations in general, and also relook at if the VN merkle proof in the QC is necessary. Not in scope for this PR so happy with these changes.

File: dan_layer/consensus/src/block_validations.rs [hunk 3]
```diff
@@ -127,32 +127,40 @@ pub async fn check_quorum_certificate<TConsensusSpec: ConsensusSpec>(
...
+    if qc.is_genesis() {
+        // This is potentially dangerous. There should be a check
+        // to make sure this is the start of the chain.
+
         return Ok(());
     }
-    if candidate_block.height() < candidate_block.justify().block_height() {
+    if candidate_block.height() < qc.block_height() {
```
Comment (sdbondi) [inline dan_layer/consensus/src/block_validations.rs:137]: Should be <=

File: integration_tests/tests/features/claim_fees.feature (hunk 2 above)
Comment (stringhandler) [reply integration_tests/tests/features/claim_fees.feature:89]: Ah ok, thanks. I've added a line to the test to also check that a non-account holder can't claim the fees

File: Cargo.toml [hunk 4]
```diff
@@ -249,26 +250,26 @@ panic = 'abort'
 # seen in debug mode. Panicking at this time is better than silently using the wrong value.
 overflow-checks = true
 
-# Make a copy of this code, uncomment and replace account and my-branch with the name of your fork and the branch you want to temporarily use
+## Make a copy of this code, uncomment and replace account and my-branch with the name of your fork and the branch you want to temporarily use
```
Comment (sdbondi) [inline Cargo.toml:253]: Please revert these changes

**Review by sdbondi — APPROVED** [review]
utACK

Looks good, some nice improvements to cucumbers.
